controller/tests/array_action/ds8k/test_array_mediator_ds8k.py

```diff
@@ -534,18 +534,45 @@ def test_create_snapshot_already_exist(self):
     def test_create_snapshot_failed_with_incorrect_id(self):
         self.client_mock.get_volume.side_effect = InternalServerError("500", message="BE7A0005")
         with self.assertRaises(array_errors.IllegalObjectID):
-            self.array.create_snapshot("volume_id", "test_name", pool=None)
+            self.array.create_snapshot("volume_id", "test_name")
 
-    def test_create_snapshot_success(self):
+    def _prepare_mocks_for_create_snapshot_with_success(self):
         volume = self._prepare_mocks_for_create_snapshot()
         self.client_mock.create_volume.return_value = volume
         self.client_mock.get_volume.return_value = volume
         self.client_mock.create_flashcopy.return_value = self.flashcopy_response
         self.client_mock.get_flashcopies.return_value = self.flashcopy_response
-        snapshot = self.array.create_snapshot("volume_id", "target_volume", pool=self.volume_response.pool)
+        return volume
+
+    def test_create_snapshot_success(self):
+        volume = self._prepare_mocks_for_create_snapshot_with_success()
```

Comment:
I think we should stop letting "prepare" methods directly return values.
consider passing it via something like `self.fake_volume`

Reply:
I changed it for correct (bug fixed and now checked) and prettier create_snapshot tests. plz review it again, sorry for the trouble.